src/runtime/runtime_internal.h

```diff
@@ -217,15 +217,20 @@ ALWAYS_INLINE T reinterpret(const U &x) {
     memcpy(&ret, &x, min(sizeof(T), sizeof(U)));
     return ret;
 }
+}  // namespace
 
-extern WEAK_INLINE int halide_malloc_alignment();
-extern WEAK_INLINE void halide_abort();
-
-void halide_thread_yield();
-
-}  // namespace Internal
+// A namespace for runtime modules to store their internal state
+// in. Should not be for things communicated between runtime modules,
+// because it's possible for them to be compiled with different c++
+// name mangling due to mixing and matching target triples.
+namespace Halide {
+namespace Runtime {
+namespace Internal {
+// Empty
+}
 }  // namespace Runtime
 }  // namespace Halide
+using namespace Halide::Runtime::Internal;
```

Comment:
If we can't safely use this to communicate between modules, should we / can we use anonymous namespaces for this purpose instead, to ensure it won't happen by accident?

Reply:
I considered it, but that also implies a linkage class, and we're being very careful with linkage right now. I'm not sure if it would be safe.